R/convert-from-datacamp.R

```diff
@@ -1,44 +1,86 @@
-
-datacamp_to_learnr <- function(.repo) {
+#' datacamp_to_learnr
+#'
+#'
+#' @export
+datacamp_to_learnr <- function(cur_path=NULL, rename=NULL, author_name="Default", has_data=T) {
   # TODO: Check if working dir is a repo
   # TODO: Check if any uncommitted files are present
   # TODO: Check that the necessary files exist
   # TODO: Check if a "original materials" git branch exists
 
-# options(usethis.quiet = TRUE)
+options(usethis.quiet = TRUE)
+
+if(is.null(cur_path)){
+  cur_path=getwd()
+}
 #
 # # For accidental sourcing... -_-
 # stop()
 #
 # # Checking if project is under git and if changes are not committed
-# if (git2r::in_repository())
-#     stop("Please make that the files are under Git version control.")
-# if (!is.null(git2r::status()$unstaged))
-#     stop("Please make sure to commit changes before running.")
+if (!in_repository(cur_path))
+    stop("Please make that the files are under Git version control.")
+if (length(status(cur_path)$unstaged)!=0)
+    stop("Please make sure to commit changes before running.")
 #
 # # Listing all files -------------------------------------------------------
-#
-# all_files <- dir_ls(here(), recursive = TRUE, type = "file", all = TRUE)
-# proj_name <- str_subset(all_files, "\\.Rproj$")
-#
+
+## needs to check to see if there are dashes in the folder name
+
+pkgname <- tail(strsplit(cur_path, "/")[[1]], 1)
+if(grepl("-", pkgname)){
+  if(is.null(rename)){
+    pkgname <- gsub("-", "", pkgname)
+  } else {
+    pkgname <- rename
+  }
+
+  new_path <- strsplit(cur_path, "/")[[1]]
+  new_path <- new_path[-length(new_path)]
+  # new_path <- new_path[-1]
+  new_path <- paste(new_path, collapse=.Platform$file.sep)
+  new_path <- file.path(new_path, pkgname)
+  file.rename(cur_path, new_path)
+  cur_path <- new_path
+}
+setwd(cur_path)
```

Comment:
We'll have to use `withr` and not `setwd()` inside of a package, since we can't have side effects. If you want to take a stab at using `withr`, check out http://withr.r-lib.org/ otherwise I can fix it.

Reply:
I'm leaving this empty for now since this speaks to the issue of workflow that you opened an issue on.